### PR TITLE
Implement dot option

### LINKED ISSUE #135: Action does not match folders starting with dot (.)
Actual:
* if I have a change in a folder `src/.deploy` and matcher for `src/**/*` it would not be matched.

Expected:
* folders starting with dot should be matched as well.

It seems the issue is related to how Minimatch works. See https://github.com/isaacs/minimatch/issues/30 . Consider passing `{dot: true}` by default.

### PR DESCRIPTION
**Pull request by kachkaev**
Closes #135

### CODE REVIEW

File: README.md
Comment (kachkaev) [inline README.md:70]: This line is not necessary and is actually confusing. We ended up with four config lines instead of two (before `dot`)

```
"area: packages":
  - "packages/*"
  - "packages/.*"
  - "packages/**/*"
  - "packages/**/.*"
```

File: README.md
Comment (kachkaev) [inline README.md:122]: What about `path/to/folder/sub/.folder/hello.txt`. Will it match? 🤔 If not, what would be the right set of globs?

**Comment by kachkaev** [comment]
@pje 🙏

File: README.md
Comment (jdufresne) [inline README.md:123]: WDYT about dropping the configuration option and making `dot` always true. 

If someone _really_ wants to ignore dot files, that is another glob expression, right? (Something already supported.) IMO, if we can solve the same problem with less configuration it will be more user friendly and more likely to "do what I mean".

Comment (kachkaev) [reply README.md:123]: That would be a breaking change. It’s safer to introduce the option, then wait for a few months, then change the default in a major release, then wait again and finally remove the option at all, if necessary.

Comment (jdufresne) [reply README.md:123]: Yes, I agree it changes the behavior, but perhaps a reasonable one. We could also bump the major version to play it safe with expectations. Ultimately, it is not my and it is only a suggestion as an end user.

**Comment by ivoanjo** [comment]
+1 Thanks for fixing this, and it would be great if this was merged in. I work at Datadog and we're using this in our repositories (see https://github.com/DataDog/dd-trace-rb/pull/2291/).

File: README.md
Comment (kachkaev) [inline README.md:60]: Ref: https://github.com/DataDog/dd-trace-rb/pull/2291#discussion_r986042860

**Comment by kachkaev** [comment]
@MaksimZhukov @panticmilos @marko-zivic-93 🙏

**Comment by panticmilos** [comment]
hi @kachkaev, we will take a look at this PR :)

**Comment by smorimoto** [comment]
When will this be fixed? It's a fundamental issue, and it's unbelievable that you're still left to fix it.

**Review by dsame — APPROVED** [review]

**Comment by smorimoto** [comment]
Is there anyone on the actions-runtime team who can review this changes?

**Comment by panticmilos** [comment]
Hi @smorimoto, we will review it in the following days, thank you for your patience.

**Comment by smorimoto** [comment]
Great! I hope this "fix" is released soon 🙂

**Comment by smorimoto** [comment]
Status?

**Comment by smorimoto** [comment]
@panticmilos hi?

**Comment by kachkaev** [comment]
PR reviews in open-source repos may take time; this is normal. If you are blocked, feel free to configure your codebase with [a forked action](https://github.com/kachkaev/labeler/tree/dot-option), which does support `dot: true`:

```yaml
name: "Pull Request Labeler"
on:
  - pull_request_target

jobs:
  triage:
    runs-on: ubuntu-22.04

    steps:
      - uses: kachkaev/labeler@a7cc8a62ef0fcdcf8ff1516986c74c7f46ef6fa4
        ## @todo replace with actions/labeler@v4 (or newer) when this PR is merged:
        ## https://github.com/actions/labeler/pull/316
        with:
          dot: true
          repo-token: "${{ secrets.GITHUB_TOKEN }}"
```

If this PRs gets futher updates, you can replace `a7cc8a62ef0fcdcf8ff1516986c74c7f46ef6fa4` with the most recent PR commit you can find on this page.

**Comment by smorimoto** [comment]
I totally get it, but sometimes things don't move forward unless we push it. This was particularly evident in my experience working with the GHA platform 🙂

**Comment by kachkaev** [comment]
@IvanZosimov, @panticmilos I just synced my branch with the latest `main`, tests do pass locally. How can I help with getting this PR in?

File: src/labeler.ts
Comment (kachkaev) [reply src/labeler.ts]: Haha, thanks for spotting this @AndreiLobanovich – great attention to detail! I have moved the bracket and have updated the PR overall. The conflicts are resolved and `npm run build && npm run test` works locally.

Happy to make further changes to this PR, if needed!

Comment (AndreiLobanovich) [reply src/labeler.ts]: @kachkaev , seems like rebuild is needed, dist check has failed

File: src/labeler.ts
Comment (MaksimZhukov) [inline src/labeler.ts:19]: We can use the `getBooleanInput` core function to parse a boolean input. Also `false` is the default value, so I think we can get rid of it.
```suggestion
    const dot = core.getBooleanInput('dot');
```

Comment (kachkaev) [reply src/labeler.ts]: I removed `{required: false}` in , thanks! https://github.com/actions/labeler/pull/316/commits/54d434dac50244b6843390a0d6987c951785b0f3

As of `getBooleanInput, it is currently not present in `main`. It was added there earlier, but then reverted in https://github.com/actions/labeler/pull/564. What’s the best way forward here?

Comment (MaksimZhukov) [reply src/labeler.ts]: Thanks for the explanation! This change was reverted and then merged into the `v5.0.0-beta.1` branch (in scope of [this PR](https://github.com/actions/labeler/pull/574)) as the [new release](https://github.com/actions/labeler/releases/tag/v5.0.0-alpha.1) is pre-release. 

Please leave `getInput` for the `sync-labels` input because fixing it is a breaking change and it was fixed in the new major release. As for the `dot` input, please use `getBooleanInput`, because currently it does not work properly (`!!'false'` = `true`).

Comment (kachkaev) [reply src/labeler.ts]: Oh I see! I replaced `getInput` with `getBooleanInput` in https://github.com/actions/labeler/pull/316/commits/71d2484daaf26523ee79447d458c78a3a1b122ee, only for the `dot` option.

**Comment by kachkaev** [comment]
What else are we missing before the PR can be merged? 🙏

**Review by AndreiLobanovich — APPROVED** [review]

File: README.md
Comment (AndreiLobanovich) [inline README.md:43]: @kachkaev, one last thing before merging. This action actually uses [minimatch](https://www.npmjs.com/package/minimatch). Could you please change the README.md?

Comment (kachkaev) [reply README.md]: You are right, well-spotted! I addressed this in https://github.com/actions/labeler/pull/316/commits/d40596e5db2302f8bd180870610f0059a6e3ace2

**Review by MaksimZhukov — APPROVED** [review]

**Review by IvanZosimov — APPROVED** [review]

**Comment by smorimoto** [comment]
I'm glad that this has finally been merged. Thank you for everyone's efforts! 🙂

**Comment by kachkaev** [comment]
[v4.1.0](https://github.com/actions/labeler/releases/tag/v4.1.0) is out yay! If anyone has been using my fork, please switch back to the official labeler action:

```diff
-      - uses: kachkaev/labeler@012b89238e3fa57d7af8ee028f02be6d421f184f
-       ## @todo replace with actions/labeler@v4 (or newer) when this PR is merged:
-       ## https://github.com/actions/labeler/pull/316
+     - uses: actions/labeler@v4
        with:
          dot: true
          configuration-path: .github/labeler.yaml
          repo-token: '${{ secrets.GITHUB_TOKEN }}'
```

I suggest to make `dot: true` the default value in actions/labeler@v5 (https://github.com/actions/labeler/issues/575#issuecomment-1576485690) – happy to create a follow-up PR if there is a consensus.

**Comment by kachkaev** [comment]
The `dot` option is `true` by default in v5 🎉
Release notes: https://github.com/actions/labeler/releases/tag/v5.0.0

**Pull request merged**
